### PR TITLE
Added configuration to CSV importer that allows to skip lines

### LINKED ISSUE #377: Option to Ignore/skip rows
While importing the data, it would be nice to skip/ignore rows since Banks always garbage filler lines above and below the actual data that we care.

### LINKED ISSUE #2157: [Feature] Skip lines when importing CSV
### Verified feature request does not already exist?

- [X] I have searched and found no existing issue

### 💻

- [ ] Would you like to implement this feature?

### Pitch: what problem are you trying to solve?

Importing transactions via CSV works flawlessly, until you have a CSV which has some sort of header followed by a line of column headers and than followed by a transaction per line.

### Describe your ideal solution to this problem

In the import dialogue for CSV there is an option to skip N 1st lines of the input file.

### Teaching and learning

_No response_

### PR DESCRIPTION
**Pull request by Horizon0156**
First of all: Thanks for the great work on Actual ❤️ 

As I'm a lazy guy and my bank is bloating up CSV exports with additional instructions on top, I've added a simple, optional configuration value that will remove 1-n additional lines from the top of the CSV file. Therefore, the user does not need to modify the bank export before importing it to Actual.

This was also requested by other users:
Fixes #377 
Fixes #2157 

This requires #3232 to be merged as well, as this is fixing the CSV import in the latest version in general.

### Additional details:
This is how a CSV export looks for my bank (ING). Ignore the language, it's just about the structure 😄

```
Umsatzanzeige;Datei erstellt am: 11.08.2024 14:54

IBAN;XX 1234 1234 1234 1234 12
Kontoname;Girokonto
Bank;ING
Kunde;XXXXXXX
Zeitraum;11.07.2024 - 11.08.2024
Saldo;1.111,11;EUR

Sortierung;Datum absteigend

In der CSV-Datei finden Sie alle bereits gebuchten Umsätze. Die vorgemerkten Umsätze werden nicht aufgenommen, auch wenn sie in Ihrem Internetbanking angezeigt werden.

Buchung;Valuta;Auftraggeber/Empfänger;Buchungstext;Verwendungszweck;Saldo;Währung;Betrag;Währung
...
```
Without manually removing the first 13 lines, I would not be able to run the importer. Now  we have the following configuration field, that defaults to 0. 
Bildschirmfoto 2024-08-11 um 15 06 04

When the "Skip lines" parameter is not adjusted (=0), no logic will be applied to the actual parsing.

### CODE REVIEW

**Comment by youngcw** [comment]
can you add a test csv to try with?  The data can be made up if you want.

**Comment by Horizon0156** [comment]
> can you add a test csv to try with? The data can be made up if you want.

Sure, please find a stripped / anonymized example with 4 transactions attached. As already mentioned in the PR description, this is an export from ING DiBa. They are adding an information block at the top with information about the reference account. 
[Export.csv](https://github.com/user-attachments/files/16575974/Export.csv)

This is a German account and therefore the heading (starting in line 13) are also German. 

FYI
Buchung = Date 
Valuta = Date (When it's actually booked)
Auftraggeber/Empfänger = Payee
Buchungstext = Booking Type
Verwendungszweck = Note
Betrag = Amount
Währung = Currency

**Comment by youngcw** [comment]
something seems broken.  Im not seeing the csv fields row and I cant get any data to show at all.

**Comment by Horizon0156** [comment]
That's why I've mentioned that #3232  needs to be merged as well, as this was fixing the CSV import in the latest version in general. This was merged by @MatissJanis  couple of hours ago. The Netlify build that was created when I filed this PR was not including the merge.

**Comment by youngcw** [comment]
the netlify gets updated every time you add a new commit, so it should be in there now

**Comment by Horizon0156** [comment]
It works with an empty account but not with an account that already has transaction. 
image

Give me a sec. Had this this morning as well while checking the general import issue and might already have the solution...

**Comment by Horizon0156** [comment]
Yep, found two places where an error is raised when a new CSV structure is imported to an account with transactions / previous imports. This was not caused by the new configuration property and happened even without my changes. 

Maybe this is also the issue mentioned in the other CSV import issue ticket. 

Please have a look. I'm new to this repo and today was my first work on it so I can't say what else happened to import in the past days.

Anyhow: While writing this Netlify updated and it seems to work on exisiting bank accounts now

**Comment by youngcw** [comment]
Seems to function correctly now.  Can you look into why the tests are failing?

**Comment by Horizon0156** [comment]
> Seems to function correctly now. Can you look into why the tests are failing?

Done.

**Review by youngcw — APPROVED** [review]

**Comment by matt-fidd** [comment]
@Horizon0156 thanks for this work! There's one small conflict to fix before we can merge

**Comment by Horizon0156** [comment]
> @Horizon0156 thanks for this work! There's one small conflict to fix before we can merge

That one was not there yesterday 😄  `master` is merged and conflict is solved.

**Review by matt-fidd — APPROVED** [review]
Thank you!

**Comment by benquan** [comment]
Thanks for this! 
Had also requested this on #2950 !!

File: packages/desktop-client/src/components/modals/ImportTransactions.jsx [hunk 1]
```diff
@@ -1282,6 +1283,10 @@ export function ImportTransactions({ options }) {
         );
         break;
       }
+      if (trans.payee == null || !(trans.payee instanceof String)) {
+        console.log(`Unable·to·parse·payee·${trans.payee || '(empty)'}`);
+        break;
+      }
```
Comment (UnderKoen) [inline packages/desktop-client/src/components/modals/ImportTransactions.jsx:1289]:
```suggestion
if (trans.payee_name == null || trans.payee_name !== "string") {
    console.log(`Unable·to·parse·payee·${trans.payee_name || '(empty)'}`);
    break;
}
```

**Pull request merged**